RFC-0001-row-id.md

```diff
@@ -0,0 +1,80 @@
+# **RFC 1 for Presto**
+
+## Row ID
+
+Proposers
+
+* Brian Dahmen
+* Devang Shah
+* Elliotte Rusty Harold
+* Serge Druzkin
+* Sreeni Viswanadha
+
+## Related Issues
+
+* Row ID support #22078
+* Binary IDs for Hive partitions #22103
+* Binary rowID type #22053
+* Start adding row\_id hidden column to Hive #22008
+
+
+## Summary
```

Comment:
I would like to see more discussion on what part of this falls on the connector and what part falls on the engine.  As I'm reading this, it seems all of these concepts can be scoped purely to the connector--is that right?

Reply:
Let me ask you two questions to see if I can understand your points better:

1. If it's all in the connector, then does `SELECT $row_id FROM table_name` still work? That is, can we still expose the $row_id pseudo-column?
2. Can the optimizer still optimize queries that use $row_id?

I think the answer to both questions is yes, but I want to make sure.

Reply:
1. Yes, `SELECT $row_id FROM table_name` will still work.  Note that `$row_id` is defined within the connector, and uses a type defined by the connector.  That is, there are no code additions in `presto-main`, `presto-common`, or `presto-spi` to facilitate this column name and type.
2. Yes, the optimizer can still optimize queries that use `$row_id`.  These columns can be identified by the connector as being unique identifiers, and as such the constraints framework may optimize them, the same as other columns that contain unique values. https://github.com/prestodb/presto/issues/16413. I think it's important that the way the optimizations are applied be general, as for example I noted below, the $row_id concept doesn't seem applicable to Hudi, whereas constraints are applicable to Hudi, Iceberg positions, the $row_id concept mentioned here, and are broadly applicable to any unique column for any data sources.

Reply:
I'm wondering if we should split this proposal into two parts:

1) Exposing $row_id in a Hive-specific, not general way.  While I worry that exposing this without guaranteeing ACID constraints is a questionable decision, I think we could make progress there (e.g., by somehow making sure it can't be persisted, or by making whatever minimal plumbing changes are required in the Hive connector, but only PrismConnector (internal) exposes it).  
2) Talking about how to get the optimizer to leverage unique columns in general, and debating the best path forward for that.

I think if the proposal and code changes were limited to 1, we could make quick progress.  I just worry 2 requires a deeper design, and we really need to settle on how to do that, as it seems we're thinking of two parallel directions there, and we shouldn't put out code that does this two different ways.